Amplify/Categories/Auth/Models/AuthUser.swift

```diff
@@ -5,10 +5,13 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+/// Defines the protocol for an auth user
 public protocol AuthUser {
 
+    /// User name of the auth user
```

Comment:
Should call out that the meaning of this field has different meanings depending on setups in the plugin. Frankly, I'd probably just discuss the different values this might have for different user pools setups, or at least point to Auth docs that discuss it.

Also, we should have Auth docs that discuss it. :) Probably updating https://docs.amplify.aws/lib/auth/overview/q/platform/ios#authentication-with-aws would be a good place to discuss the differences in behaviors